math/genvector/inc/Math/GenVector/DisplacementVector3D.h

```diff
@@ -333,13 +335,26 @@ namespace ROOT {
       Scalar Perp2() const { return fCoordinates.Perp2();}
 
       /**
-         return unit vector parallel to this
+         return unit vector parallel to this (scalar)
       */
-      DisplacementVector3D  Unit() const {
-        Scalar tot = R();
+      template< typename SCALAR = Scalar >
```

Comment:
I personally find it clearer enabling it as a template template default arg.
`template <typename SCALAR = Scalar, typename = std::enable_if< std::is_arithmetic<SCALAR>::value>` and have a clear return type of the function.